fuel-vm/src/interpreter/contract.rs

```diff
@@ -365,6 +345,63 @@ impl<'vm, S, Tx> TransferCtx<'vm, S, Tx> {
     }
 }
 
+fn get_contract_id_from_memory(
```

Comment:
You can actually replace all new functions with `CheckedMemConstLen` and `read`, it should be shorter and more readable and will unify the code

<img width="825" alt="image" src="https://github.com/FuelLabs/fuel-vm/assets/18346821/1c97656a-3b07-45f2-a15e-f921ab631d6a">

Reply:
Went with
```
fn get_address_from_memory(
    offset: Word,
    memory: &[u8; MEM_SIZE],
) -> Result<Address, RuntimeError> {
    get_from_memory::<{ Address::LEN }, Address>(offset, memory)
}

fn get_from_memory<const SIZE: usize, T>(
    offset: Word,
    memory: &[u8; MEM_SIZE],
) -> Result<T, RuntimeError>
where
    T: From<[u8; SIZE]>,
{
    let range = CheckedMemConstLen::<SIZE>::new(offset)?;
    let bytes = range.read(memory);
    let value = T::from(*bytes);
    Ok(value)
}
```
Which I'm happy with.

BUT

I wish that `LEN` was a associated `const` for some trait and then we could let the type checker fill in the generics for us:
```
fn get_from_memory<T>(
    offset: Word,
    memory: &[u8; MEM_SIZE],
) -> Result<T, RuntimeError>
where
    T: WithSize
    T: From<[u8; <T as WithSize>::SIZE]>,
{
    let range = CheckedMemConstLen::<<T as WithSize>::SIZE>::new(offset)?;
    let bytes = range.read(memory);
    let value = T::from(*bytes);
    Ok(value)
}
```

I would add a trait for it, but then we'd need to import that trait everywhere we want the `LEN` and that's not worth it.

Reply:
I find this even cleaner, as it's modular yet still readable:

```rust
let contract_id = ContractId::from(read_bytes(self.memory, contract_id_addr)?);
````

Reply:
Agreed. I added that change. I'm keeping the helper functions though because I prefer my code to read declaratively.

Reply:
Hmm, but I think it is already descriptive enough: "Read bytes from memory with offset as ContractId"=)

```rust
let contract_id = ContractId::from(read_bytes(memory, offset)?);
```

Reply:
Okay. 2v1. I'll change it.